src/core/Observer.ts

```diff
@@ -336,36 +399,123 @@ export class Observer {
     return rtDocument;
   }
 
+  /**
+   * Start subscription or pulling on the collection
+   *
+   * @internal
+   */
+  watchCollection (index: string, collection: string) {
+    return this.mode === 'realtime'
+      ? this.resubscribe(index, collection)
+      : this.restartPulling();
+  }
+
+  private restartPulling (): Promise<void> {
+    return this.clearPullingTimer()
+      .then(() => {
```

Comment:
why `then`?

Reply:
The `clearPullingTimer` method return a promise to be consistent with `resubscribe` and be able to [do that](https://github.com/kuzzleio/sdk-javascript/pull/676/files/8216c50d305395153623af41c578d72dc794df46#diff-af8a751f1013cdb6a3c52a0edca81545b07cf0e586cfcaee87268fc977f6d239R234)
![image](https://user-images.githubusercontent.com/4447392/148374611-b725571d-7c77-449a-b0fd-461488a93beb.png)

But you're right, it was lazyness and this method don't need to return a promise